sktime/datatypes/tests/test_vectorize.py

```diff
@@ -0,0 +1,321 @@
+
+    Fixtures parameterized
+    ----------------------
+    scitype : str - scitype of fixture
+    mtype : str - mtype of fixture
+    fixture_index : int - index of fixture tuple with that scitype and mtype
+    """
+    # collect fixture tuples here
+
+    sci_mtype_tuples = _generate_scitype_mtype_combinations()
+
+    sci_mtype_index_tuples = []
+
+    for tuple_j in sci_mtype_tuples:
+        scitype = tuple_j[0]
+        mtype = tuple_j[1]
+        n_fixtures = len(get_examples(mtype=mtype, as_scitype=scitype))
+
+        for i in range(n_fixtures):
+            sci_mtype_index_tuples += [(scitype, mtype, i)]
+
+    return sci_mtype_index_tuples
+
+
+def pytest_generate_tests(metafunc):
+    """Test parameterization routine for pytest.
+
+    Fixtures parameterized
+    ----------------------
+    scitype : str - scitype of fixture
+    mtype : str - mtype of fixture
+    fixture_index : int - index of fixture tuple with that scitype and mtype
+    iterate_as : str - level on which to iterate over
+    """
+    # we assume all four arguments are present in the test below
+
+    fixturenames = set(metafunc.fixturenames)
+
+    if set(["scitype", "mtype", "fixture_index"]).issubset(fixturenames):
+        keys = _generate_scitype_mtype_fixtureindex_combinations()
+
+        ids = []
+        for tuple in keys:
+            ids += [f"{tuple[0]}-{tuple[1]}-fixture:{tuple[2]}"]
+
+        # parameterize test with from-mtpes
+        metafunc.parametrize("scitype,mtype,fixture_index", keys, ids=ids)
+
+    elif set(["scitype", "mtype"]).issubset(fixturenames):
+        keys = _generate_scitype_mtype_combinations()
+
+        ids = []
+        for tuple in keys:
+            ids += [f"{tuple[0]}-{tuple[1]}"]
+
+        # parameterize test with from-mtpes
+        metafunc.parametrize("scitype,mtype", keys, ids=ids)
+
+    if "iterate_as" in fixturenames:
+        metafunc.parametrize("iterate_as", ["Panel", "Series"])
+
+
+def test_construct_VectorizedDF(scitype, mtype, fixture_index):
+    """Test VectorizedDF constructs with valid arguments.
+
+    Fixtures parameterized
+    ----------------------
+    scitype : str - scitype of fixture
+    mtype : str - mtype of fixture
+    fixture_index : int - index of fixture tuple with that scitype and mtype
+    """
+    # retrieve fixture for checking
+    fixture = get_examples(mtype=mtype, as_scitype=scitype).get(fixture_index)
+
+    # iterate as Series, without automated identification of scitype
+    VectorizedDF(X=fixture, iterate_as="Series", is_scitype=scitype)
+
+    # iterate as Series, with automated identification of scitype
+    VectorizedDF(X=fixture, iterate_as="Series", is_scitype=None)
+
+    # iterate as Panel, can only do this if scitype is hierarchical
+    if scitype == "Hierarchical":
+        VectorizedDF(X=fixture, iterate_as="Panel", is_scitype=None)
+
+
+def test_construct_VectorizedDF_errors(scitype, mtype, fixture_index):
```

Comment:
VecotrizedDf shouldnt be capatilized

Reply:
well, but it's the name of the class that is tested. Should I still lower-case it?

Reply:
Yea always lower case in function names even refering to a class

Reply:
ok, will change